match/lib/match/storage/gitlab/client.rb

```diff
@@ -0,0 +1,79 @@
+
+module Match
+  module Storage
+    class GitLab
+      class Client
+        def initialize(api_v4_url:, project_id:, job_token: nil, private_token: nil)
```

Comment:
Should we raise an error if they try to set both job and private token?

Reply:
What do you think about logging a warning instead? I'd prefer not to fail the CI job in this case, but logging a warning to let the user know what is happening could be a good option. In this case we would default to the `JOB_TOKEN`.